streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/processor/CustomProcessorInfo.java

```diff
@@ -222,7 +222,7 @@ private ComponentUISpecification getCustomProcessorUISpecification (ComponentUIS
             }
         }
         result.setFields(fields);
-        return result;
+        return (result.getFields().size() == 0 ? null : result);
```

Comment:
nit: use fields.isEmpty()
Q: will it cause NPE at the caller? maybe better to return an Optional.

Reply:
Changed to isEmpty.
No, it wont cause a NPE. Its a private method and the only call is in this class that relies on it returning null.